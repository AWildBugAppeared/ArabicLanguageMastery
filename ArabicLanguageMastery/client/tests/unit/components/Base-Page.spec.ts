/* eslint-disable  @typescript-eslint/no-explicit-any */

// Imports
import Vue from 'vue';
import Vuetify from 'vuetify';

// Utilities
import { createLocalVue, mount, Wrapper } from '@vue/test-utils';

import BasePage from '@/components/Base-Page.vue';

describe('BasePage.vue', () => {
  const localVue = createLocalVue();
  let wrapper: Wrapper<Vue & { [key: string]: any }>;
  let vuetify: Vuetify;

  const mountWrapper = () => {
    wrapper = mount(BasePage, {
      localVue,
      vuetify,
      slots: {
        heading: '<h1>Some text in the heading slot',
        default: '<div>Some text in the default slot',
      },
    });
  };

  beforeEach(() => {
    vuetify = new Vuetify({
      mocks: {
        $vuetify: {},
      },
    });
    Vue.use(Vuetify);
    document.body.setAttribute('data-app', 'true');
  });

  afterEach(() => {
    wrapper.destroy();
    jest.resetModules();
    jest.clearAllMocks();
  });

  describe('render', () => {
    it('should render base page', async () => {
      mountWrapper();
      await wrapper.vm.$nextTick();

      expect(wrapper.html()).toMatchSnapshot();
    });
  });
});
