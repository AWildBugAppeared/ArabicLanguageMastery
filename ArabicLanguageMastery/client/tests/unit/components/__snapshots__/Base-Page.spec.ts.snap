// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BasePage.vue render should render base page 1`] = `
"<div class=\\"container page-container\\">
  <h1 class=\\"text-h3 text-center\\">
    <h1>Some text in the heading slot</h1>
  </h1>
  <div>Some text in the default slot</div>
</div>"
`;
